docs/product/specs/forms.md

```diff
@@ -0,0 +1,466 @@
+	- ![select-producer.png](./img/forms/select-producer.png)
+	- ![select-director.png](./img/forms/select-director.png)
+	- ![select-genre.png](./img/forms/select-genre.png)
+	- Admin can configure whether or not they want to allow end users to add new records.
+		- If they allow the '+ Add New' option is shown in the dropdowns.
+		- Admins can configure the text shown.
+	- This is a record selector shown in a dropdown. Only the summaries are shown in the list.
+	- There will be a small Pagination section when the entires are more than 20.
+	- The search area contains the columns that are part of the record summary.
+	- These columns could be from any related joinable table, as configured by the record summary.
+	- The admin cannot choose searchable columns if they are not part of the record summary configuration. The Record summary configuration is the only source here.
+	- The admin can customize the labels for these columns.
+- **Adding new records in related tables via fk fields:**
+	- ![form-new-entry-fk-field.png](./img/forms/form-new-entry-fk-field.png)
+	- End users can choose to add new records for Producers, Directors, and Genres.
+	- Notice that the "Director" field shows the field "Person" for the column `person_id` which is an a FK for `People`.
+	- The user could still add a nested entry for the `People` table using the "Person" dropdown.
+		- ![form-new-nested-fk-entry.png](./img/forms/form-new-nested-fk-entry.png)
+	- "Genres", however shows fields from the `Genre` table directly, since `Movie_Genre_mapping` is a mapping table with just a reverse-fk.
+		- It adds entries to both `Genre` and `Movie_Genre_mapping`.
+
+## Scope
+
+### Initial release of "Forms":
+1. Select form fields from base table.
+1. Select form fields from related tables - both fk links and reverse-fk-links.
+1. Publicly sharing a form for end-users to fill.
+1. Customizable styling and success/failure messages.
+1. Preview forms while creating/editing.
+1. Basic Permissions.
+	- Each form is owned by a role.
+1. Redirection after submission.
+	- Allow redirection to a custom endpoint after successful form submission. This would happen after showing the success message, with a timer.
+1. Basic form-level validations.
+	- Marking fields as "required" on the form to disallow empty fields, while the underlying columns allow NULL & empty strings.
+
+### To consider in future releases:
+1. Granular permissions management for forms.
+1. Restrictions on filling forms.
+	- Login required,
+	- Password,
+	- Captchas, etc.,
+1. Email notifications after a form is submitted.
+1. Creating an 'exploration' from a 'form' to view de-normalized data filled via forms.
+1. Perform DDL operations while building form.
+	- Being able to create a new table directly in the form view.
+	- Being able to create new columns directly in the form view.
+1. Advanced form-level validations.
+1. Multi-language support for end-users filling the form.
+1. Customizable Logos.
```

Comment:
I just want to make some small suggestions here:

```suggestion
1. Customizable logos, banners, and accent color.
1. Conditional logic to display form fields based on sibling field values
```

Reply:
Thanks! I've committed the suggestion.